docs/modules/ROOT/pages/explanations/templating.adoc

```diff
@@ -0,0 +1,40 @@
+= Templating
+
+Some fields of the CRDs allow to use https://golang.org/pkg/text/template[Go templates] which are rendered by the operator.
+You can find some explanations on how to use them in here.
+
+
+== Cluster Template
+
+The tenant field `.spec.clusterTemplate` <<references/crds-html.adoc#syn.tools/v1alpha1.TenantSpec,defines>> a template which is used to set default values for clusters of this tenant.
+All _string_ fields within this allow to use Go templates which are rendered by the operator.
+As input data to the template rendering, the instance of the cluster on which the template is being applied on can be used.
+This data is a Go type defined https://github.com/projectsyn/lieutenant-operator/blob/54dd0a6540843903586db52a57c2bebb89cfec2b/pkg/apis/syn/v1alpha1/gitrepo_types.go#L113[here].
+
+Since it's a Go type, only https://golang.org/ref/spec#Exported_identifiers[exported fields] (uppercase) can be accessed.
+
+Some examples of how to access different values:
+
+[source,yaml]
+----
+apiVersion: syn.tools/v1alpha1
+kind: Tenant
+metadata:
+  name: t-aezoo6
+spec:
+  displayName: Big Corp.
+  gitRepoTemplate:
+    apiSecretRef:
+      name: vshn-gitlab
+    path: syn-dev/customers
+    repoName: t-aezoo6
+  clusterTemplate:
+    facts:
+      name: '{{ index .Annotations "syn.tools/name"}}'
+      creation: '{{ .CreationTimestamp }}'
+    gitRepoTemplate:
+      apiSecretRef:
+        name: secret-{{ .Spec.TenantRef.Name }}
+      path: cluster-catalogs/{{ index .Spec.Facts "cloud"}}
```

Comment:
Please explicitly state that the `index` function must be used when accessing dicts.